control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -93,10 +93,6 @@ func (b *meshGatewayBuilder) consulDataplaneContainer(containerConfig v2beta1.Ga
 				Name:  envDPCredentialLoginMeta,
 				Value: "pod=$(POD_NAMESPACE)/$(DP_PROXY_ID)",
 			},
-			{
-				Name:  envDPServiceNodeName,
-				Value: "$(NODE_NAME)-virtual",
-			},
```

Comment:
The flags in consul-dataplane [indicate](https://github.com/hashicorp/consul-dataplane/blob/b80f2bf4a4876789ae0b524275b8201e692b7177/cmd/consul-dataplane/main.go#L59C1-L61C57) that this envar is deprecated and should use `proxy-node-name` instead; however, the docs for `proxy-node-name` indicate that it's ignored for the v2 API.